piptools/scripts/compile.py

```diff
@@ -444,6 +444,16 @@ def cli(
     # Proxy with a LocalRequirementsRepository if --upgrade is not specified
     # (= default invocation)
     if not upgrade and os.path.exists(output_file.name):
+        output_file_is_empty = os.path.getsize(output_file.name) == 0
+        if upgrade_install_reqs and output_file_is_empty:
```

Comment:
```suggestion
    output_file_exists = os.path.exists(output_file.name)
    output_file_is_empty = os.path.getsize(output_file.name) == 0
    if not upgrade and output_file_exists:
        if upgrade_install_reqs and output_file_is_empty:
```

I'm still fine with the `output_file_exists` variable though.

Reply:
`os.path.getsize()` will throw a `FileNotFoundError` if `output_file` does not exist. That's why @webknjaz proposed 

    output_file_is_empty = not output_file_exists or os.path.getsize(output_file.name) == 0

I think that's fine too, but short-circuiting that way is not the most intuitive path to read.

Reply:
> `os.path.getsize()` will throw a `FileNotFoundError` if `output_file` does not exist.

I'm not quite sure what that has to do with the separate `output_file_exists` variable, can you explain?